### PR TITLE
Deprecate Unwrap method in favour of compactMap

### PR DESCRIPTION
**Pull request by HugoSay**
RxSwift 5 added the CompactMap operator which overlaps with the current RxSwiftExt `unwrap()` method. 

Should we deprecate the use of the unwrap method in favour of the new compactMap naming?

### CODE REVIEW

File: Source/RxSwift/unwrap.swift [hunk 1]
```diff
@@ -16,8 +16,17 @@ extension ObservableType {
...
-        return self.filter { $0 != nil }.map { $0! }
+        return self.compactMap({ $0 })
+    }
+
+    /**
+     Takes a sequence of optional elements and returns a sequence of non-optional elements, filtering out any nil values.
+
+     - returns: An observable sequence of non-optional elements
+     */
+    public func compactMap<T>() -> Observable<T> where Element == T? {
```
Comment (freak4pc) [inline Source/RxSwift/unwrap.swift]: We probably shouldn't add this. Just deprecate the original. Thanks :)

Comment (HugoSay) [reply Source/RxSwift/unwrap.swift]: Ok, I'll remove it then. 
Also I'll wait for the next release of RxSwift including compactMap on sharedSequence Single and Maybe In order to deprecate all the old code at once !
(https://github.com/ReactiveX/RxSwift/pull/1978)

File: Source/RxSwift/unwrap.swift [hunk 2]
```diff
@@ -16,8 +16,8 @@ extension ObservableType {
 
      - returns: An observable sequence of non-optional elements
      */
-
-    public func unwrap<Result>() -> Observable<Result> where Element == Result? {
```
Comment (freak4pc) [inline Source/RxSwift/unwrap.swift:20]: why did you change the signature? It's best to keep the same as it was, just add the deprecation, nothing else.
